qf-faucet-server/src/main.rs

```diff
@@ -0,0 +1,258 @@
+use std::str::FromStr;
+
+use subxt;
+use subxt::{OnlineClient, PolkadotConfig};
+use subxt_signer::bip39::Mnemonic;
+use warp::{reject::Reject, Filter};
+use subxt_signer::sr25519::dev;
+use subxt_signer::sr25519;
+use std::env;
+use hex;
+use clap::Parser;
+use subxt::utils::{MultiAddress, AccountId32};
+use rusqlite::{Connection, Result as SqlResult};
+use chrono::Utc;
+
+
+#[subxt::subxt(runtime_metadata_path = "metadata.scale", derive_for_all_types = "PartialEq, Eq")]
+pub mod polkadot {}
+
+const RPC_URL: &str = "wss://dev.qfnetwork.xyz/socket";
+const FAUCET_AMOUNT: u128 = 20_000_000_000;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Serialize)]
+struct DripRequest {
+    address: String,
+}
+
+#[derive(Debug)]
+struct TransferError;
+
+impl Reject for TransferError {}
+
+#[derive(Debug)]
+struct SomeError;
+
+impl Reject for SomeError {}
+
+#[derive(Debug)]
+struct StorageError;
+
+impl Reject for StorageError {}
+
+fn get_db() -> SqlResult<Connection> {
+    let conn = Connection::open("transfers.db")?;
+    conn.execute(
+        "CREATE TABLE IF NOT EXISTS transfers (
+            id INTEGER PRIMARY KEY,
+            address TEXT NOT NULL,
+            amount INTEGER NOT NULL,
+            timestamp INTEGER NOT NULL,
+            tx_hash TEXT NOT NULL
+        )",
+        [],
+    )?;
+    Ok(conn)
+}
+
+fn store_transfer(address: &str, amount: u64, tx_hash: &str) -> SqlResult<()> {
+    let conn = get_db()?;
+    conn.execute(
+        "INSERT INTO transfers (address, amount, timestamp, tx_hash) VALUES (?1, ?2, ?3, ?4)",
+        (address, amount, Utc::now().timestamp(), tx_hash),
+    )?;
+    Ok(())
+}
+
+fn can_transfer(address: &str, timeout: Option<u64>) -> SqlResult<bool> {
+    let conn = get_db()?;
+    let now = Utc::now().timestamp() as u64;
+    let two_hours_ago = now - (timeout.unwrap_or_else(|| 120) * 60);
+
+    let count: i64 = conn.query_row(
+        "SELECT COUNT(*) FROM transfers 
```

Comment:
Hello,

Seems no indexes or mat views, hence this database would be degraded and stop working after some time.

Also seems we store data forever it does not look right.